### PR TITLE
Adds an Escape Menu

### PR DESCRIPTION
**Pull request by Mothblocks**
Needs #72907

## About The Pull Request

https://user-images.githubusercontent.com/35135081/213997512-b15ef784-33af-4b0d-9092-65ced2660ee1.mp4

I forgot to show it off but the "Space Station 13" text is the station name and will update automagically, also suicide/ghost close the menu now

## Why It's Good For The Game

Part of long term vision to make the stat/verb panel redundant. Information that is currently locked to the stat panel, but is still useful to know at some point, can now exist here.

Admin help is really important, so it's here now. Nothing moved over yet, to be clear, but this will be the Ideal place for it.

Makes the "ghost" verb no longer accessible only through the verb panel. "Suicide" is also no longer a completely hidden (??) verb.

## Changelog

:cl:
add: Added a new menu that shows on Escape.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Might split out adminhelp stuff into its own PR because it's really annoying

**Comment by JohnFulpWillard** [comment]
escape menus generally have the option to leave the game and close it entirely. Since we don't need those at all, maybe we can have Server Hop there instead?

**Comment by Time-Green** [comment]
Incredibly cursed but pretty sick

**Comment by Mothblocks** [comment]
@JohnFulpWillard i can add server hop as a replacement for leave body when you're dead

**Comment by Mothblocks** [comment]
later though

File: code/modules/escape_menu/escape_menu.dm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	new /datum/escape_menu(src)
+
+#define PAGE_HOME "PAGE_HOME"
+#define PAGE_LEAVE_BODY "PAGE_LEAVE_BODY"
+
+/datum/escape_menu
+	/// The client that owns this escape menu
+	var/client/client
+
+	VAR_PRIVATE
```
Comment (Mothblocks) [inline code/modules/escape_menu/escape_menu.dm:28]: i did this because its hideous when you apply it to everything individually

**Comment by Iamgoofball** [comment]
@JohnFulpWillard @Mothblocks we should have a quit button on the escape menu regardless imho

Also, if respawn is enabled on a server, a. return to lobby button.

**Comment by Mothblocks** [comment]
> we should have a quit button on the escape menu regardless imho

maybe. i just think adding it makes it now ambiguous with "leave body", which i already had to clear with a few people was intuitive

**Comment by Mothblocks** [comment]
we are merge skewed but my ci is failing because of c&d

**Comment by IndieanaJones** [comment]
Holy shit I've never thought of an escape menu being a bizarre and cool feature, but here we are.

**Comment by Iamgoofball** [comment]
Leaving your body is more akin to going to spectator mode imho

**Comment by itseasytosee** [comment]
Holy Hell

File: code/modules/escape_menu/home_page.dm [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+	if (hovered)
+		maptext = "<u>[maptext]</u>"
+
+/atom/movable/screen/escape_menu/home_button/admin_help
+	VAR_PRIVATE
+		current_blink = FALSE
+		is_blinking = FALSE
+		last_blink_time = 0
+
+		blink_interval = 0.4 SECONDS
```
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/escape_menu/home_page.dm:140]: i mentioned it earlier--VAR_PRIVATE is extremely long and noisy and doing this makes it actually readable. vars are already always relatively pathed anyway

[the remark this answers is not in the document]
Comment (ZephyrTFA) [reply code/modules/escape_menu/home_page.dm:140]: I know, saw the comment after I made this comment but decided to keep the comment so you didn't get a ghost email

**Comment by Serxule** [comment]
who needs ss14 to be a remake?

we can just turn ss13 into ss14

**Comment by Tastyfish** [comment]
> Part of long term vision to make the stat/verb panel redundant. 

Is there also a goal to remove the menu bar?

**Comment by Serxule** [comment]
> > Part of long term vision to make the stat/verb panel redundant.
> 
> Is there also a goal to remove the menu bar?

do you mean the command bar at the bottom right of the screen?

**Comment by Mothblocks** [comment]
> Is there also a goal to remove the menu bar?

which one

**Comment by Tastyfish** [comment]
> > Is there also a goal to remove the menu bar?
> 
> which one

The actual menu bar at the top with File, Help, etc.

**Comment by Mothblocks** [comment]
probably not until we're really bored

File: code/_onclick/hud/screen_object_holder.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/// A helper instance that will handle adding objects from the client's screen
+/// to easily remove from later.
+/datum/screen_object_holder
+	VAR_PRIVATE
```
Comment (LemonInTheDark) [inline code/_onclick/hud/screen_object_holder.dm:4]: I was promised it would remain locked away, never to be shown to mortal man

Comment (Iamgoofball) [reply code/_onclick/hud/screen_object_holder.dm:4]: violation of our contributing standards

Comment (Mothblocks) [reply code/_onclick/hud/screen_object_holder.dm:4]: who promised you that

Comment (ZephyrTFA) [reply code/_onclick/hud/screen_object_holder.dm:4]: I use VAR_PRIVATE all the time

**Pull request merged**